consensus/src/pipeline/body_processor/body_validation_in_context.rs

```diff
@@ -18,13 +19,25 @@ impl BlockBodyProcessor {
     }
 
     fn check_block_transactions_in_context(self: &Arc<Self>, block: &Block) -> BlockProcessResult<()> {
-        let (pmt, _) = self.window_manager.calc_past_median_time(&self.ghostdag_store.get_data(block.hash()).unwrap())?;
+        // TODO: this is somewhat expensive during ibd, as it incurs cache misses.
+        let pmt = Lazy::new(|| {
+            let (pmt, _) = self
+                .window_manager
+                .calc_past_median_time(&self.ghostdag_store.get_data(block.hash()).unwrap())
+                .expect("expected header processor to have checked error case");
```

Comment:
Actually not 100% about this `expect`

Reply:
I can also init after the quick check, and propagate error back in the function.